### PR TITLE
Typescript bindings for XP Program

### PR DESCRIPTION
**Pull request by moshthepitt**
## Changes

This PR adds Typescript bindings for the XP program.

The program itself is in this companion PR: https://github.com/staratlasmeta/star-atlas-programs/pull/80

## Checklist

- [ ] UAT
- [ ] Passes final QA checks

### CODE REVIEW

**Comment by bravetarget** [comment]
Task linked: [CU-257en2b Factory calls for XP Program](https://app.clickup.com/t/257en2b)

File: src/xp.ts [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+};
+
+/** Base Params for instructions */
+interface BaseParams {
+  connection: Connection /** the Solana connection object */;
+  programId: web3.PublicKey /** Deployed program ID for the XP program */;
+}
+
+/** Params for Init instruction */
+export interface InitXpVarsParams extends BaseParams {
```
Comment (bzierk) [inline src/xp.ts:126]: Is there value in creating these interfaces? It looks like each is used once with its corresponding function

File: src/xp.ts [hunk 2]
```diff
@@ -0,0 +1,746 @@
...
+      accounts: {
+        admin,
+        xpVarsAccount: xpVarsAccountKey,
+        systemProgram: web3.SystemProgram.programId,
+      },
+    }),
+  ];
+
+  return {
+    xpVarsAccount: xpVarsAccountKey,
```
Comment (bzierk) [inline src/xp.ts:153]: Is it convention to return PDAs with the instructions?

Comment (bzierk) [reply src/xp.ts:153]: I'm fine either way but I think we should settle on one or the other (the other being somethign like `Promise<web3.TransactionInstruction>` for the sake of consistency across Factory. We don't want Jim grumpier than he already is.

Comment (visualjc) [reply src/xp.ts:153]: @bzierk / @moshthepitt - really doesn't matter to me if you return just the instruction or the instruction + the account as a tuple; what does matter is that the function should define it's return type in the function delecration.

```
export const initXpVarsIx = async ({
  admin,
  connection,
  programId,
}: InitXpVarsParams): Promise<{ account:string, instruction: web3.TxInstruction}>
```

That is way more helpful to us on the web side.

File: src/xp.ts (hunk 1 above)
Comment (visualjc) [reply src/xp.ts:126]: @bzierk - yes, it helps with TypeScript intellisense and readability to me.

**Comment by moshthepitt** [comment]
This has been superseded by https://github.com/staratlasmeta/factory/pull/71

**Pull request closed without merging**